### DISCUSSION TITLE
run duration has been increased after upgrade to the latest version

### DISCUSSION THREAD

**Discussion by carpet92** [JanDeDobbeleer/oh-my-posh#4934] [General]
I've been using the `v18.27.1` version of oh-my-posh for a long time. Today, after upgrading to the latest `v19.29.1`, I noticed that the run time has been increased to 50 ms, whereas v18.27.1 took about 10 ms.

After that I downgraded back to the 18.27.1 using:

```
$ winget uninstall JanDeDobbeleer.OhMyPosh
$ winget install --version 18.27.1 JanDeDobbeleer.OhMyPosh -s winget
```

A couple of screenshots:

-----

![18 27 1](https://github.com/JanDeDobbeleer/oh-my-posh/assets/20873402/124504b2-9d4e-4b02-b973-f74b99f4a999)

-----

![19 29 1](https://github.com/JanDeDobbeleer/oh-my-posh/assets/20873402/bf3f44e1-ad69-42f5-94b8-5fc9587ca08f)

-----

- PowerShell 7.4.2
- Windows Terminal Preview Version: 1.21.1272.0
- Added all exclusions for Windows Defender (for pwsh.exe, windowsterminal.exe, wt.exe, oh-my-posh, for all related processes etc.)

**Comment by JanDeDobbeleer** [comment]
@carpet92 I'm going to need the output of those calls to see what's taking more time. You can pipe to clip using `oh-my-posh debug --plain | clip` and paste here.

**Comment by carpet92** [comment]
@JanDeDobbeleer when I run the `oh-my-posh debug --plain | clip` command, it produces no output for me. So, I just uploaded the results of `oh-my-posh debug > debug_18.27.1.txt` and `oh-my-posh debug > debug_19.29.1.txt` to gist:

- debug_18.27.1.txt https://gist.githubusercontent.com/carpet92/c479d74773019b439e3df82729d39c72/raw/4eaf525b6632f1f433fa693aadf18523360a6bf6/debug_18.27.1.txt
- debug_19.29.1.txt https://gist.githubusercontent.com/carpet92/c479d74773019b439e3df82729d39c72/raw/4eaf525b6632f1f433fa693aadf18523360a6bf6/debug_19.29.1.txt

**Reply by JanDeDobbeleer** [reply]
The difference seems to come from `shell.go:Getenv(TERM_PROGRAM)` which changed from `0s` to `31.1914ms` which makes no sense. Unless `go` changed the way it fetches this information for the first call, in which case it doesn't matter which env var we try to get. Doesn't seem related to something inside oh-my-posh itself, as everything else stayed the same in first glace.

**Reply by JanDeDobbeleer** [reply]
I updated `go` to see if this issue persists, keep me in the loop on latest.

**Reply by JanDeDobbeleer** [reply]
BTW, the theory would be that go1.22.0 introduced this, meaning that [v19.10.0](https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v19.10.0) would be OK and [v19.11.0](https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v19.11.0) would have introduced this.

**Reply by carpet92** [reply]
@JanDeDobbeleer

- debug_19.31.0.txt
https://gist.githubusercontent.com/carpet92/870c9fffcce38211e865567cb4302d50/raw/44aa8fb52b5e35c445b3c42728e442fde25141b3/debug_19.31.0.txt

**Reply by JanDeDobbeleer** [reply]
@carpet92 check, so the latest `go` doesn't fix it. Can you try the two versions I listed above?

**Reply by carpet92** [reply]
@JanDeDobbeleer sure. Both `19.10.0` and `19.11.0` show a run time of about 10 ms.

- debug_19.10.0.txt
https://gist.githubusercontent.com/carpet92/f182df5a4e991a402d6475eae037802d/raw/2ba1fcec432ddc511113468db3837b1b31861ff5/debug_19.10.0.txt

- debug_19.11.0.txt
https://gist.githubusercontent.com/carpet92/f182df5a4e991a402d6475eae037802d/raw/2ba1fcec432ddc511113468db3837b1b31861ff5/debug_19.11.0.txt

TBH there is no visible difference (at first glance at least) between loading times of these two versions (I mean 18.27.1 and latest 19.3*), but I prefer 9 ms so perhaps I'm going to stay on `v19.11.0`

**Reply by JanDeDobbeleer** [reply]
Hmm. This makes no sense. These functions haven't changed at all, that's the only influencing factor...

BTW, if you notice the difference in 10 vs 40 milliseconds, something else if off IMHO.

**Reply by carpet92** [reply]
@JanDeDobbeleer 
> if you notice the difference in 10 vs 40 milliseconds

Actually no. I just eventually ran `oh-my-posh debug` after the upgrade to `19.3` and noticed it

**Reply by JanDeDobbeleer** [reply]
@carpet92 log wise, yes, experience wise that shouldn't make a difference.

**Reply by carpet92** [reply]
@JanDeDobbeleer isn't this problem reproduced in your setup? If not, then the problem may be on my side. Perhaps silly question, but ... can't this be caused by an old version of `go` installed locally? I have `go 1.17.8` and I didn't use it since this version so it's such an old.

**Reply by JanDeDobbeleer** [reply]
@carpet92 it should be compiled so no. But I need to update my local VM to reproduce. I'm on my mac ATM.

**Reply by carpet92** [reply]
@JanDeDobbeleer just a heads-up. I tried to iteratively install oh-my-posh version by version and check run duration, and found that the discussion topic started to appear since `v19.16.0`. Log: [debug_19.16.0.txt](https://gist.githubusercontent.com/carpet92/88cadb5873e9ace67712d286ff335984/raw/526cdf707d1d7fbe269f9570f0725aa90462501f/debug_19.16.0.txt)

**Reply by JanDeDobbeleer** [reply]
@carpet92 so it is the upgrade to go1.22. I'm going to file a bug on their end for this one.
